### PR TITLE
Python 3: bytes/str updates for sockets

### PR DESCRIPTION
**Pull request by hereischen**
Signed-off-by: Haotong Chen <hachen@redhat.com>

### CODE REVIEW

File: virttest/rss_client.py [hunk 1]
```diff
@@ -135,6 +135,8 @@ def close(self):
         self._socket.close()
 
     def _send(self, sr, timeout=60):
+        if isinstance(sr, six.string_types):
```
Comment (chunfuwen) [inline virttest/rss_client.py]: Maybe we can use existed astring.is_text to do conditional judgement first ,then do encode().

**Comment by hereischen** [comment]
Just a note here, `sr` could be `str` or `bytes` in different contexts.

13:01:32 ERROR|   File "/home/staf-kvm-devel/workspace/avocado-vt/virttest/rss_client.py", line 142, in _send
    self._socket.sendall(sr)
13:01:32 ERROR| TypeError: a bytes-like object is required, not 'str'
++++++++++++++++++++++++++++++++++++++++++++++++++
13:09:05 ERROR|   File "/home/staf-kvm-devel/workspace/avocado-vt/virttest/rss_client.py", line 142, in _send
    self._socket.sendall(sr.encode())

13:09:05 ERROR| AttributeError: 'bytes' object has no attribute 'encode

**Comment by hereischen** [comment]
passed on Python3

File: virttest/qemu_monitor.py [hunk 2]
```diff
@@ -1469,9 +1471,9 @@ def _read_objects(self, timeout=READ_OBJECTS_TIMEOUT):
 
     def _send(self, data):
         """
-        Send raw data without waiting for response.
+        Send raw bytes data without waiting for response.
```
Comment (luckyh) [inline virttest/qemu_monitor.py]: Should also take care of `HumanMonitor._send()`.

File: virttest/qemu_monitor.py [hunk 3]
```diff
@@ -1631,7 +1633,7 @@ def cmd(self, cmd, args=None, timeout=CMD_TIMEOUT, debug=True, fd=None):
                     self._socket, fd, json.dumps(cmdobj) + "\n")
                 self._log_lines(str(cmdobj))
             else:
-                self._send(json.dumps(cmdobj) + "\n")
+                self._send(json.dumps(cmdobj).encode() + b"\n")
```
Comment (luckyh) [inline virttest/qemu_monitor.py]: Should also take care of `QMPMonitor.cmd_raw()` and `HumanMonitor.cmd()`.

File: virttest/guest_agent.py [hunk 4]
```diff
@@ -209,7 +209,7 @@ def _read_objects(self, timeout=READ_OBJECTS_TIMEOUT):
         """
         if not self._data_available():
             return []
-        s = ""
+        s = b""
```
Comment (luckyh) [inline virttest/guest_agent.py]: There might be some issues at line 231 and 234.

Comment (hereischen) [reply virttest/guest_agent.py]: updated, thanks

**Comment by luckyh** [comment]
Closing in favor of https://github.com/avocado-framework/avocado-vt/pull/1599

**Pull request closed without merging**